front/src/pages/Home/Home.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { todosAPI } from '../../api/todos-api';
+import { TodosArr } from '../../types';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootStateType } from '../../redux/store';
+import { setTodos } from '../../redux/todo-reducer';
+import { setIsFetching } from '../../redux/app-reducer';
+import { AxiosError } from 'axios';
+import { addMessage } from '../../redux/messages-reducer';
+import styles from './Home.module.scss';
+
+type useQueryType = {
+  error: AxiosError | null
+  [propName: string]: any
+}
+
+const Home = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const { todos } = useSelector((state: RootStateType) => state.todos)
+
+  const {
+    isFetching,
+  }: useQueryType = useQuery(
+    ['todos'],
+    todosAPI.getAllTodos,
+    {
+      onSuccess: (todosArr: TodosArr ) => {
+        dispatch(setTodos(todosArr));
+      },
+      onError: (error) => {
+        dispatch(addMessage({
+          text: error?.message || '',
+        }));
+      },
+      onSettled: () => {
+        dispatch(setIsFetching(false));
+      },
+    }
+  );
+
+  // set up loader
+  if (isFetching) {
+    dispatch(setIsFetching(true));
+  }
+
+  return (
+    <div className="container">
+      { todos?.length > 0 &&
```

Comment:
better not overuse "?.". Here todos is always either `[]` or non-empty array. So just use `todos.length > 0`